### PR TITLE
time picker: dropdown mode retemplating ability

### LINKED ISSUE #4656: Time picker component fails on dropdown mode in combination with igxTimePickerTemplate modifications
## Description  
If igx-time-picker [mode]="dropdown" and the igxTimePickerTemplate directive is used to modify the layout of the igx-time-picker, the time picker crashes, and does not render any values. 

 * igniteui-angular version: 7.2.6
 * browser: Chrome

## Steps to reproduce  

1. Open the following stackblitz application containing the code i'm trying to run https://stackblitz.com/edit/angular-xepcq4
2. Click the suffix icon to open the dialog 
3. Click the suffic icon again

## Result  
After suffix is clicked a second time, values will be rendered, and can be edited, but dialog cannot be closed and values will not be saved to the input field

## Expected result  
The dropdown opens and render values on first click, when the user clicks outside the bounds of the dialog or exits in any other way, the dialog is closed and specified value are displayed in the input field.

### PR DESCRIPTION
**Pull request by SAndreeva**
Closes #4656

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts [hunk 1]
```diff
@@ -1092,6 +1089,59 @@ export class IgxTimePickerComponent implements
         return currentVal;
     }
 
+    private _initializeContainer() {
+        if (this.value) {
+            const foramttedTime = this._formatTime(this.value, this.format);
```
Comment (bkulov) [inline projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts]: Typo `foramttedTime`.

File: projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts [hunk 2]
```diff
@@ -1147,97 +1197,56 @@ export class IgxTimePickerComponent implements
...
+     * @ViewChild('timePicker', { read: IgxTimePickerComponent }) picker: IgxTimePickerComponent;
+     * picker.openDialog();
      * ```
      */
     public openDialog(timePicker: IgxTimePickerComponent = this): void {
-
-        if (this.collapsed) {
-            this.collapsed = false;
-
+        // debugger;
```
Comment (bkulov) [inline projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts]: Delete // debugger;

File: projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts [hunk 3]
```diff
@@ -1092,6 +1089,59 @@ export class IgxTimePickerComponent implements
...
+            this.selectedAmPm = this._ampmItems[3];
+        }
+
+        this._prevSelectedHour = this.selectedHour;
+        this._prevSelectedMinute = this.selectedMinute;
+        this._prevSelectedAmPm = this.selectedAmPm;
+
+        this._onTouchedCallback();
+
+        this._updateHourView(0, 7);
```
Comment (bkulov) [inline projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts]: Please add constants for this 'magic' numbers.

File: projects/igniteui-angular/src/lib/time-picker/time-picker.common.ts [hunk 4]
```diff
@@ -29,7 +28,7 @@ export interface IgxTimePickerBase {
     scrollHourIntoView(item: string): void;
     scrollMinuteIntoView(item: string): void;
     scrollAmPmIntoView(item: string): void;
-    hideOverlay(): void;
+    close(): void;
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/time-picker/time-picker.common.ts]: Please add a migration.

**Pull request merged**